packages/discovery-react-components/src/utils/document/processDoc.ts

```diff
@@ -329,6 +329,10 @@ function setupSectionParser(
 
           if (tagName === TABLE_TAG) {
             sectionHtml.push('</article>');
+            sectionHtml[0] = sectionHtml[0].replace(
```

Comment:
This concerns me. Are we sure it will always be index 0?

What about making use of `openTagIndices` instead, since that's already working? In `onopentag`, when `tagName === TABLE_TAG`, push to `openTagIndices`. Then later in `onclosetag`, do a pop and update, again in the `tagName === TABLE_TAG` section (which is then followed by an existing pop and updated).